data_manager.py

```
@@ -0,0 +1,281 @@
+    def load_city_restaurants(self, city: str, vibe: Optional[str] = None, 
+                            min_rating: float = 4.3) -> pd.DataFrame:
+        """Load restaurants for a specific city and optionally filter by vibe"""
+        
+        # Check cache first
+        cache_key = f"{city}_{vibe}_{min_rating}" if vibe else f"{city}_all_{min_rating}"
+        cache_file = self.cache_dir / f"{cache_key}.pkl"
+        
+        if cache_file.exists():
+            try:
+                return pd.read_pickle(cache_file)
+            except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)